test/KoreBuild.Tasks.Tests/DownloadFileTests.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.IO;
+using BuildTools.Tasks.Tests;
+using Microsoft.Build.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+using Task = System.Threading.Tasks.Task;
+
+namespace KoreBuild.Tasks.Tests
+{
+    public class DownloadTests
+    {
+        private readonly ITestOutputHelper _output;
+
+        public DownloadTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact]
+        public async Task ItDownloadAFile()
+        {
+            var packages = new[]
+            {
+                new TaskItem("Newtonsoft.Json", new Hashtable { ["Version"] = "9.0.1", ["Source"] = "https://api.nuget.org/v3/index.json"} ),
+            };
+
+            var expectedPath = Path.Combine(AppContext.BaseDirectory, "microsoft.com.html");
+            var task = new DownloadFile
+            {
+                Uri = "https://microsoft.com/",
```

Comment:
`http://example.org/index.html` (I'd rather we not use TLS here to avoid flakiness due to server cert expiration)